src/editor/codemirror/language-server/common.ts

```diff
@@ -0,0 +1,25 @@
+import { Facet } from "@codemirror/state";
+import { EditorView } from "@codemirror/view";
+import { LanguageServerClient } from "../../../language-server/client";
+
+// What's the best way to setup a singleton required facet?
```

Comment:
Perhaps it's best to acknowledge that these could be null and provide helpers to read them?